pkg/mover/chart.go

```diff
@@ -236,17 +285,116 @@ func namespacedPath(fullpath, chartName string) string {
 }
 
 /*
-Move executes the Chart relocation which includes
+  Move performs the relocation.
+
+A regular move executes the Chart relocation which includes
 
 1 - Push all the images to their new locations
 
 2 - Rewrite the Helm Chart and its subcharts
 
 3 - Repackage the Helm chart as toChartFilename
+
+A save to an offline tarball bundle will:
+
+1 - Drop all images to disk, with the original chart (unpacked) and hints file
+
+2 - Package all in a single compressed tarball
 */
 func (cm *ChartMover) Move() error {
+	if cm.targetOfflineTar != "" {
+		return cm.saveOfflineBundle()
+	}
+	return cm.moveChart()
+}
+
+func (cm *ChartMover) saveOfflineBundle() error {
 	log := cm.logger
 
+	tarPath, err := os.MkdirTemp("", "offline-tarball-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory to build tar: %w", err)
+	}
+	log.Printf("Writing chart at %s/...\n", cm.chart.Metadata.Name)
+	if err := writeChart(cm.chart, filepath.Join(tarPath, cm.chart.Metadata.Name)); err != nil {
+		return fmt.Errorf("failed archiving chart %s: %w", cm.chart.Name(), err)
+	}
+	if err := packImages(tarPath, cm.imageChanges, cm.logger); err != nil {
+		return fmt.Errorf("failed archiving images: %w", err)
+	}
+	log.Printf("Writing hints file %s...\n", HintsFilename)
+	if err := os.WriteFile(filepath.Join(tarPath, HintsFilename), cm.rawHints, defaultTarPermissions); err != nil {
+		return fmt.Errorf("failed to write hints file: %w", err)
+	}
+	log.Printf("Packing all as tarball %s...\n", cm.targetOfflineTar)
+	if err := tarDirectory(tarPath, cm.targetOfflineTar); err != nil {
+		return fmt.Errorf("failed to tar bundle as %s: %w", cm.targetOfflineTar, err)
+	}
+	return os.RemoveAll(tarPath)
+}
+
+func writeChart(chart *chart.Chart, targetDir string) error {
+	if err := os.MkdirAll(targetDir, defaultTarPermissions); err != nil {
+		return fmt.Errorf("failed to create target chart %s: %w", targetDir, err)
+	}
+	for _, file := range chart.Raw {
+		dir := filepath.Dir(file.Name)
+		dirPath := filepath.Join(targetDir, dir)
+		if err := os.MkdirAll(dirPath, defaultTarPermissions); err != nil {
+			return fmt.Errorf("failed to create path %s: %w", dirPath, err)
+		}
+		if err := os.WriteFile(filepath.Join(targetDir, file.Name), file.Data, defaultTarPermissions); err != nil {
+			return fmt.Errorf("failed to write chart's inner file %s: %v", file.Name, err)
+		}
+	}
+	return nil
+}
+
+func packImages(archivePath string, imageChanges []*internal.ImageChange, logger Logger) error {
+	imagesTarball := filepath.Join(archivePath, "images.tar")
+	tagToImage := map[name.Tag]v1.Image{}
+	for _, change := range imageChanges {
+		app := change.ImageReference.Context().Name()
+		version := change.ImageReference.Identifier()
```

Comment:
and rename version for tag?

Reply:
nope, we have `tag` a bit bellow meaning "name and version", I think version is better here. At the end of the day is the version (tag) part.